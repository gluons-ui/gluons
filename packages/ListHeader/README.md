## Mandala UI ListHeader Component
Mandala is a React component library that utilizes a declarative CSS library for styling.

### Installation
Check the [MonoRepo README](https://github.com/mandala-ui/mandala) for installation.

### Description
A header for the top of lists that matches the style and props

### Usage
if using the whole library:
```js
import { ListHeader } from 'mandala';
```
if only using this component:
```js
import ListHeader from '@mandala-ui/list-header';
```
### Props
* `children` - what needs to be in the header, usually one h element.
* `indented` - adds padding for a contained list
* `lineColor` - the color of the bottom border
* `lined` - set this prop to add a bottom border
* `lineWidth` - width of the bottom border of the header

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| children | node     | null         | -          |
| indented | boolean  | false        | -          |
| lineColor| string   | 'white'      | -          |
| lined    | boolean  | false        | -          |
| lineWidth| number   | 0            | -          |
| ordered  | boolean  | false        | -          |

#### TODO:
- [ ] tight, normal, high for vertical padding.
- [ ] infer `lined` from lineColor prop existence

![mandala-bottom](https://user-images.githubusercontent.com/1824267/38281372-32114e5c-375f-11e8-9fbe-e5191b96429c.png)
