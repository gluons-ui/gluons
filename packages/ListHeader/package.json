{
  "author": "Lucas K Allmon",
  "description": "Mandala UI Kit ListHeader Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user",
    "ux"
  ],
  "license": "MIT",
  "main": "lib/ListHeader.js",
  "module": "es/ListHeader.js",
  "name": "@mandala-ui/list-header",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "1.0.0"
}
