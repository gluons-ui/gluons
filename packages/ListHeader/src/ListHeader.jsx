import React from 'react';
import PropTypes from 'prop-types';

const ListHeader = ({
  children,
  indented,
  lineColor,
  lined,
  lineWidth }) => {
  const hasLine = `${lined ? `bb ${lineWidth ? `bw${lineWidth}` : ''}` : ''}`;
  const indents = `${indented ? 'ph3' : ''}`;
  return (
    <div className="list-header">
      <div className={`b--${lineColor} ${hasLine} pb2`}>
        <span className={`${indents}`}>
          {children || null}
        </span>
      </div>
    </div>
  );
};

ListHeader.propTypes = {
  children: PropTypes.node,
  indented: PropTypes.bool,
  lineColor: PropTypes.string,
  lined: PropTypes.bool,
  lineWidth: PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
};

ListHeader.defaultProps = {
  children: null,
  indented: false,
  lineColor: 'black',
  lined: false,
  lineWidth: 0,
};

export default ListHeader;
