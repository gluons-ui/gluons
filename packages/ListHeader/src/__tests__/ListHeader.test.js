import React from 'react';
import { shallow } from 'enzyme';
import ListHeader from '../ListHeader';

describe('<ListHeader />', () => {
  it('exists', () => {
    expect(<ListHeader />).toBeDefined();
  });
  describe('Rendering', () => {
    it('renders Default', () => {
      expect(shallow(
        <ListHeader />,
      )).toMatchSnapshot();
    });
    it('renders children', () => {
      expect(shallow(
        <ListHeader>
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can render a line divider', () => {
      expect(shallow(
        <ListHeader
          lined
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can render a different color line', () => {
      expect(shallow(
        <ListHeader
          lineColor="red"
          lined
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can be indented for a contained list', () => {
      expect(shallow(
        <ListHeader
          indented
          lineColor="red"
          lined
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can be have a custom line width of (1)', () => {
      expect(shallow(
        <ListHeader
          lined
          lineWidth={1}
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can be have a custom line width of (2)', () => {
      expect(shallow(
        <ListHeader
          lined
          lineWidth={2}
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can be have a custom line width of (3)', () => {
      expect(shallow(
        <ListHeader
          lined
          lineWidth={3}
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can be have a custom line width of (4)', () => {
      expect(shallow(
        <ListHeader
          lined
          lineWidth={4}
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
    it('can be have a custom line width of (5)', () => {
      expect(shallow(
        <ListHeader
          lined
          lineWidth={5}
        >
          <h3>The Top of a List</h3>
        </ListHeader>,
      )).toMatchSnapshot();
    });
  });
});
