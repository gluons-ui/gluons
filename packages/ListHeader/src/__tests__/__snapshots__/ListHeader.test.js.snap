// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListHeader /> Rendering can be have a custom line width of (1) 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering can be have a custom line width of (2) 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering can be have a custom line width of (3) 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering can be have a custom line width of (4) 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering can be have a custom line width of (5) 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering can be indented for a contained list 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering can render a different color line 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering can render a line divider 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering renders Default 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ListHeader /> Rendering renders children 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
